dockerfiles/haskell/8.8.4/Dockerfile

```diff
@@ -0,0 +1,44 @@
+# First Stage: Builder
+FROM debian:buster-slim AS builder
+
+# Install necessary dependencies for building GHC
+RUN apt-get update \
+    && apt-get install -y --no-install-recommends \
+    curl build-essential libffi-dev libgmp-dev libtinfo5 ca-certificates xz-utils \
+    && rm -rf /var/lib/apt/lists/*
+
+# Install GHCup (Haskell toolchain manager) without automatically installing the latest GHC
+RUN curl --proto '=https' --tlsv1.2 -sSf https://get-ghcup.haskell.org | bash -s -- -y --no-ghc \
+    && /root/.ghcup/bin/ghcup install ghc 8.10.7 \
+    && /root/.ghcup/bin/ghcup set ghc 8.10.7 \
+    && rm -rf /root/.ghcup/ghc/9.4.8
+
+# Clean up unnecessary build artifacts and cache
+RUN rm -rf /root/.ghcup/cache /var/lib/apt/lists/*
+
+
+# Second Stage: Final Image
```

Comment:
Why do you need two stages? Can you just install all needed packages initially and then remove any non runtime necessary packages.

Reply:
sure

Reply:
No. I specifically broke up the layers to make the image smaller. If you remember, this image used to be almost 8GB large. It is now less than 4. This is why I used a builder, `debian:buster-slim`, and repeatedly removed un-needed dependencies as I went along.